lib/cube_browser/browse.py

```diff
@@ -0,0 +1,80 @@
+import glob
+import IPython.display
+import ipywidgets
+import iris
+import matplotlib.pyplot as plt
+
+import cube_browser
+
+# Clear output, such as autosave disable notification.
+IPython.display.clear_output()
+
+# Defines the file system path for input files.
+path = ipywidgets.Text(
+    description='Path:',
+    value=iris.sample_data_path('GloSea4'),
+)
+def handle_path(sender):
+    options = glob.glob('{}/*'.format(path.value))
+    options.sort()
+    files.options = options
+# path is the widget which i want to observe
+# the first argument is the function to run on 'event'
+# names is the thing that i want know about, within path
+path.observe(handle_path, names='value')
+
```

Comment:
It might be useful to more explicitly explain the relationship between path, handle_path and path.observe.  You have done this here in between handle_path and path.observe, but to me the flow doesn't feel right.  It is not clear that path.observe causes a change in options for path.